torch/distributed/checkpoint/_async_process_executor.py

```diff
@@ -172,6 +178,8 @@ def _execute_save(
         checkpoint_request_id: _CheckpointRequestIdentifier,
         storage_writer: Optional[StorageWriter] = None,
         planner: Optional[SavePlanner] = None,
+        no_dist: bool = False,
```

Comment:
Starting to fear the Boolean trap here? Is use_collectives valid with no_dist True or should this actually be an enum?